### ISSUE TITLE
Bug in ConvOclBwdWrW53 isApplicable

### ISSUE THREAD

**Issue by JehandadKhan** [ROCmSoftwarePlatform/MIOpen#353]
The `IsApplicable` method of `ConvOclBwdWrW53` does not check the direction of the context and within the `IsApplicable` method calls the GetSolution method, which throws an exception if the convolution direction is not acceptable. 

Ideally, the `IsApplicable` method should check the direction of convolution and return false instead of throwing an exception.

**Comment by daniellowell** [comment]
Where else does this crop up?

**Comment by JehandadKhan** [comment]
In a few other solvers, I discovered this while enumerating solvers in `Fin`. Will update the issue with other solvers as well.

**Comment by JehandadKhan** [comment]
The following solvers show as applicable in the forward direction, however, they don't raise exceptions:

* ConvAsmBwdWrW3x3
* ConvOclBwdWrW2NonTunable

**Comment by atamazov** [comment]
@JehandadKhan Could you please shed some light ___how___ could I reproduce the problem?

**Comment by JehandadKhan** [comment]
@atamazov You cannot reproduce this problem using MIOpen driver since MIOpen catches exceptions occurring inside solvers and silently moves on. I discovered this using a C++ level tool that I am writing, which is specifically designed to operate on the solver level.

**Comment by atamazov** [comment]
IIRC the library never tries to use WrW solvers for Forward and Backward directions. What is the sequence of miopen API calls that may lead to this? I see two options:
- Corrupt find-db
- Missing call to `miopenConvolutionGet*Solution()` prior calling `miopenConvolution*Immediate()`

In both cases throwing an exception is an appropriate behavior. What I am missing?

**Comment by JehandadKhan** [comment]
I discovered this using C++ objects, not using the API, as follows:

```c++
// construct the context
// Get the solver object
// Call isApplicable() on the solver -> exception thrown when it should have returned false
```

Since this issue has been addressed with #354  I am closing this issue.
